core/org.wso2.carbon.ndatasource.core/src/main/java/org/wso2/carbon/ndatasource/core/DataSourceRepository.java

```diff
@@ -512,7 +512,7 @@ public boolean testDataSourceConnection(DataSourceMetaInfo dsmInfo) throws DataS
 		try {
 			return dsReader.testDataSourceConnection(DataSourceUtils.elementToString((Element)dsmInfo.getDefinition().getDsXMLConfiguration()));
 		} catch (DataSourceException e) {
-			log.error(e.getMessage(), e);
+			log.debug(e.getMessage(), e);
```

Comment:
debug logs should be added withing if condition to check whether debug logs are enabled

if (log.isDebugEnabled()) {
			log.debug();
		}